shm/tests/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 3.10)
+
+enable_testing()
+
+include_directories(mocks)
+
+set(TEST_SOURCES
+  coordinatesTest.cpp
+)
+
+add_executable(${PROJECT_NAME}-ut main.cpp ${TEST_SOURCES})
+target_compile_options(${PROJECT_NAME}-ut PRIVATE -Wall -Wextra -Wpedantic -Wconversion)
+target_link_libraries(${PROJECT_NAME}-ut PUBLIC ${PROJECT_NAME}_lib gtest_main gmock_main)
```

Comment:
I see one problem here. `${PROJECT_NAME}_lib` is defined in another subdirectory. If someone replaces the order od add_subdirectory this project will not compile. If you want to have a module structure of CMake you need to take care of those details. `{$PROJECT_NAME}_lib` should be eg. defined in the root CMakeLists.txt to avoid this issue.
And still you have `gtest_main` and `gmock_main`. It should be `gtest` and `gmock`.